pkg/controller/datavolume-controller.go

```diff
@@ -2599,3 +2623,288 @@ func GetRequiredSpace(filesystemOverhead float64, requestedSpace int64) int64 {
 func newLongTermCloneTokenGenerator(key *rsa.PrivateKey) token.Generator {
 	return token.NewGenerator(common.ExtendedCloneTokenIssuer, key, 10*365*24*time.Hour)
 }
+
+// detectCloneSize obtains and assigns the original PVC's size when cloning using an empty storage value
+func (r *DatavolumeReconciler) detectCloneSize(
+	dv *cdiv1.DataVolume,
+	pvcSpec *corev1.PersistentVolumeClaimSpec,
+	cloneType cloneStrategy) (bool, error) {
+
+	var targetSize resource.Quantity
+	sourcePvc, err := r.findSourcePvc(dv)
+	if err != nil {
+		return false, err
+	}
+	sourceSize := sourcePvc.Status.Capacity.Storage()
+
+	// Due to possible filesystem overhead complications when cloning
+	// using host-assisted strategy, we create a pod that automatically
+	// collects the size of the original virtual image with 'qemu-img'.
+	// If another strategy is used or the original PVC's volume mode
+	// is "block", we simply extract the value from the original PVC's spec.
+	if cloneType == HostAssistedClone && getVolumeMode(sourcePvc) == corev1.PersistentVolumeFilesystem {
+		// If available, we first try to get the virtual size from previous iterations
+		imgSize, available := getSizeFromAnnotations(sourcePvc)
+		if !available {
+			imgSize, err = r.getSizeFromPod(sourcePvc)
+			if err != nil {
+				return false, err
+			} else if imgSize == ImgSizeNotReady {
+				return false, nil
+			}
+		}
+		// We enable the clone-controller to skip the size comparison requirement
+		dv.Annotations[AnnPermissiveClone] = "true"
+		// If needed, inflate size with filesystem overhead
+		targetSize, err = inflateSizeWithOverhead(r.client, imgSize, pvcSpec, sourceSize)
+		if err != nil {
+			return false, err
+		}
+	} else {
+		targetSize = *sourceSize
+	}
+
+	pvcSpec.Resources.Requests[corev1.ResourceStorage] = targetSize
+	return true, nil
+}
+
+// getSizeFromAnnotations checks the source PVC's annotations and returns the requested size if it has already been obtained
+func getSizeFromAnnotations(sourcePvc *corev1.PersistentVolumeClaim) (string, bool) {
```

Comment:
When I see a function called getSizeSomething, I am expecting the return value to be a number of some kind, in this case I would expect either a Quantity or an int64 instead of a string.

Reply:
I chose to return a string so it can be more easily converted to Quantity after the overhead calculation, but `int64` would also work well. I'll change it.